game/hiscore.cc

```diff
@@ -40,45 +42,41 @@ void Hiscore::addHiscore(HiscoreItem&& item) {
 		throw std::runtime_error("No track given");
 	if (!reachedHiscore(item.score, item.songid, item.level, item.track))
 		return;
-	m_hiscore.insert(std::move(item));
-}
 
-Hiscore::HiscoreVector Hiscore::queryHiscore(std::optional<PlayerId> playerid, std::optional<SongId> songid, std::string const& track, std::optional<unsigned> max) const {
```

Comment:
While this method is very generic and could be used from many contexts, it was only being used from a single one. Accessing one of the maps was the more efficient alternative so I replaced the usage and deleted this function.